guides/predictive-analytics-with-deepnote.md

```diff
@@ -0,0 +1,119 @@
+
+
+<figure><img src="../.gitbook/assets/.predictive-analytics/datasource.png" alt=""><figcaption><p>Selecting PostreSQL</p></figcaption></figure>
+
+3. Create a new project by clicking on projects at the left panel of the screen
+
+<figure><img src="../.gitbook/assets/.predictive-analytics/create project.png" alt=""><figcaption><p>Create a new project</p></figcaption></figure>
+
+4.  Make an integration to your Hydra data warehouse by integrating PostgreSQL. Retrieve your database credentials from your Hydra dashboard. You will need Hostname, User, Password, and Database.
+
+<figure><img src="../.gitbook/assets/.predictive-analytics/Integrating postgres.png" alt=""><figcaption><p>Adding integrations</p></figcaption></figure>
+
+<figure><img src="../.gitbook/assets/.predictive-analytics/connecting postgres.png" alt=""><figcaption><p>Inputing your Hydra credentials</p></figcaption></figure>
+
+5. After establishing the connection, you can view the imported salary table within Deepnote.
+Using the SELECT statement, you can run SQL queries on the salary table, enabling you to retrieve and analyze specific data from the table.
+
+```sql
+SELECT * FROM public.salary LIMIT 100
+```
+
+<figure><img src="../.gitbook/assets/.predictive-analytics/Hydra dashboard.png" alt=""><figcaption><p>Deepnote dashboard after connection</p></figcaption></figure>
+
+6. To access the table using Python scripts, an engine has to be created with the data warehouse credentials
+
+```python
+import pandas as pd
+from sqlalchemy import create_engine
+#user = username
+#pass = password
+#server = hostname
+#database = database-name
+engine = create_engine('postgresql://user:pass@server:5432/database')
+#table_name = Salary
+#database = public
+data = pd.read_sql_table('table_name',con= engine, schema= 'database')
+```
+
+**Congratulations!** You've successfully connected Hydra with Deepnote! What next?
+
+## Making the predictions
+
+Now that the Database is connected we can use Python and SQL scripts to access the data. Using the Linear Regression model, we can predict the salary data.
+The necessary models and libraries are being imported.The linear regression model and matplotlib for our visualization are imported as shown below&#x20;
+
+```python
+#import Classes and lib needed
+import matplotlib.pyplot as plt
+from sklearn.linear_model import LinearRegression
+```
+
+The dataset is split into dependent and independent variables(X, Y),the model is trained on the dataset, and prediction on the salary table  is made.
+
+
+```python
+#defining variables X and Y
+X = data.iloc[:,0].values.reshape(-1,1)
+print(X)
+Y = data.iloc[:,1].values.reshape(-1,1)
+print(Y)
+```
+
+```python
+#creating a Linear Regression model for prediction
+model = LinearRegression()
+
+#Fitting the model on the Dataset
+model.fit(X,Y)
+
+Y_pred = model.predict(X)  # make predictions
+```
+
+Visualizations are made on the actual salary and the predicted to view the difference.
```

Comment:
Visualizations are created to show the difference between the actual salary versus the predicted salary.